File: DailyChallenges/CPP/LeetCode_452.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end());
        int ans = 0, end = points[0][1];
        for (auto& a: points){
            if (a[0] > end){
                end = a[1];
                ++ans;
            } else
                end = min(end, a[1]);
        }
        return ans+1;
    }
};

File: DailyChallenges/Scala/LeetCode_452.scala
object Solution {
    def findMinArrowShots(points: Array[Array[Int]]): Int = {
        val sps = points.sortWith((a1,a2) => a1(0)<a2(0) || (a1(0) == a2(0) && a1(1) < a2(1)))
        sps.tail.foldLeft((sps.head(1), 1))((r, p) => if (p(0)>r._1) (p(1), r._2+1) else (math.min(r._1, p(1)), r._2))._2
    }
}
